### PR TITLE
Feature argo removal

### PR DESCRIPTION
**Pull request by andrewKstrt**
# 📲 What

Argo removed from project

# 🤔 Why

With Swift.Codable we don't need Argo anymore

# 🛠 How

Swift.Codable protocol adoption for the models was implemented

# ⏰ TODO

- [ ] Push Notification Testing
- [ ] DeepLink Testing

### CODE REVIEW

File: KsApi/models/Comment.swift [hunk 1]
```diff
@@ -24,34 +23,19 @@ extension Comment: Swift.Decodable {
     self.author = try values.decode(Author.self, forKey: .author)
     self.body = try values.decode(String.self, forKey: .body)
     self.createdAt = try values.decode(TimeInterval.self, forKey: .createdAt)
-    self.deletedAt = try values.decode(TimeInterval?.self, forKey: .deletedAt)
+    // Decode a time interval so that non-positive values are coalesced to `nil`. We do this because the API
+    // sends back `0` when the comment hasn't been deleted, and we'd rather handle that value as `nil`.
+    if let value = try values.decodeIfPresent(TimeInterval.self, forKey: .deletedAt), value > 0 {
```
Comment (singhhari) [inline KsApi/models/Comment.swift:28]: Is it possible to simplify/re-factor this with a nil-coalescing operator?
e.g. `self.deletedAt = try values.decodeIfPresent(TimeInterval.self, forKey: .deletedAt)` ?? nil

File: KsApi/models/ErrorEnvelope.swift [hunk 2]
```diff
@@ -120,71 +129,70 @@ public struct ErrorEnvelope {
...
-}
 
-extension ErrorEnvelope.KsrCode: Argo.Decodable {
-  public static func decode(_ j: JSON) -> Decoded<ErrorEnvelope.KsrCode> {
-    switch j {
-    case let .string(s):
-      return pure(ErrorEnvelope.KsrCode(rawValue: s) ?? ErrorEnvelope.KsrCode.UnknownCode)
-    default:
-      return .typeMismatch(expected: "ErrorEnvelope.KsrCode", actual: j)
+  public init(from decoder: Decoder) throws {
```
Comment (singhhari) [inline KsApi/models/ErrorEnvelope.swift:174]: Also wondering if nil-coalescing would work here? Not sure how  `try` affects the use of that operator.

**Review by singhhari — APPROVED** [review]
left a few comments but otherwise lgtm

File: KsApi/models/Comment.swift (hunk 1 above)
Comment (andrewKstrt) [reply KsApi/models/Comment.swift:28]: we need to have `value > 0` check in place, and i don't know the way to make it work with nil-coalescing operator

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**